android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt

```diff
@@ -50,1235 +50,773 @@ import com.bitmovin.player.api.ui.UiConfig
-            }
-            return bufferMediaTypeConfig
-        }
+/**
+ * Converts any JS object into a `BufferMediaTypeConfig` object.
+ */
+fun ReadableMap.toBufferMediaTypeConfig(): BufferMediaTypeConfig = BufferMediaTypeConfig().apply {
+    withDouble("forwardDuration") { forwardDuration = it }
+}
 
-        /**
-         * Converts any JS object into a `BufferConfig` object.
-         * @param json JS object representing the `BufferConfig`.
-         * @return The generated `BufferConfig` if successful, `null` otherwise.
-         */
-        @JvmStatic
-        fun toBufferConfig(json: ReadableMap?): BufferConfig? {
-            if (json == null) {
-                return null
-            }
-            val bufferConfig = BufferConfig()
-            if (json.hasKey("audioAndVideo")) {
-                toBufferMediaTypeConfig(json.getMap("audioAndVideo"))?.let {
-                    bufferConfig.audioAndVideo = it
-                }
-            }
-            if (json.hasKey("restartThreshold")) {
-                bufferConfig.restartThreshold = json.getDouble("restartThreshold")
-            }
-            if (json.hasKey("startupThreshold")) {
-                bufferConfig.startupThreshold = json.getDouble("startupThreshold")
-            }
-            return bufferConfig
-        }
+/**
+ * Converts any JS object into a `BufferConfig` object.
+ */
+fun ReadableMap.toBufferConfig(): BufferConfig = BufferConfig().apply {
+    withMap("audioAndVideo") { audioAndVideo = it.toBufferMediaTypeConfig() }
+    withDouble("restartThreshold") { restartThreshold = it }
+    withDouble("startupThreshold") { startupThreshold = it }
+}
 
-        /**
-         * Converts an arbitrary [ReadableMap] to a [RemoteControlConfig].
-         *
-         * @param json JS object representing the [RemoteControlConfig].
-         * @return The generated [RemoteControlConfig].
-         */
-        private fun toRemoteControlConfig(json: ReadableMap?): RemoteControlConfig? {
-            if (json == null) return null
-            val defaultRemoteControlConfig = RemoteControlConfig()
-
-            val receiverStylesheetUrl = json.getOrDefault(
-                "receiverStylesheetUrl",
-                defaultRemoteControlConfig.receiverStylesheetUrl,
-            )
-
-            var customReceiverConfig = defaultRemoteControlConfig.customReceiverConfig
-            if (json.hasKey("customReceiverConfig")) {
-                customReceiverConfig = json.getMap("customReceiverConfig")
-                    ?.toHashMap()
-                    ?.mapValues { entry -> entry.value as String } ?: emptyMap()
-            }
+/**
+ * Converts an arbitrary [ReadableMap] to a [RemoteControlConfig].
+ */
+private fun ReadableMap.toRemoteControlConfig(): RemoteControlConfig = RemoteControlConfig().apply {
+    withString("receiverStylesheetUrl") { receiverStylesheetUrl = it }
+    getMap("customReceiverConfig")
```

Comment:
```
withMap("customReceiverConfig") { customReceiverConfig = it.toHashMap().mapValues { entry -> entry.value as String } }
```
Do we even need `toHashMap`?

Reply:
Yes, but moved it in to a helper function: 899d4f0749701ad6aba1fdd93810d1f53be2eb97